messaging-activemq/src/main/java/org/wildfly/extension/messaging/activemq/ha/ManagementHelper.java

```diff
@@ -0,0 +1,80 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+
+package org.wildfly.extension.messaging.activemq.ha;
+
+import static org.jboss.as.controller.OperationContext.Stage.MODEL;
+
+import java.util.Collection;
+import java.util.Set;
+
+import org.jboss.as.controller.AbstractAddStepHandler;
+import org.jboss.as.controller.AttributeDefinition;
+import org.jboss.as.controller.OperationContext;
+import org.jboss.as.controller.OperationFailedException;
+import org.jboss.as.controller.OperationStepHandler;
+import org.jboss.as.controller.PathAddress;
+import org.jboss.as.controller.descriptions.ModelDescriptionConstants;
+import org.jboss.as.controller.registry.Resource;
+import org.jboss.dmr.ModelNode;
+import org.wildfly.extension.messaging.activemq.ActiveMQReloadRequiredHandlers;
+import org.wildfly.extension.messaging.activemq.logging.MessagingLogger;
+
+/**
+ * @author <a href="http://jmesnil.net/">Jeff Mesnil</a> (c) 2014 Red Hat inc.
+ */
+public class ManagementHelper {
+
+    /**
+     * Create an ADD operation that can check that there is no other sibling when the resource is added.
+     *
+     * @param childType the type of children to check for the existence of siblings
+     * @param allowSibling whether it is allowed to have sibling for the resource that is added.
+     * @param attributes the attributes of the ADD operation
+     */
+    static AbstractAddStepHandler createAddOperation(final String childType, final boolean allowSibling, Collection<? extends AttributeDefinition> attributes) {
+        return new ActiveMQReloadRequiredHandlers.AddStepHandler(attributes) {
+            @Override
+            public void execute(OperationContext context, ModelNode operation) throws OperationFailedException {
+                super.execute(context, operation);
+                if (!allowSibling) {
+                    context.addStep(checkNoOtherSibling(childType), MODEL);
+                }
+            }
+        };
+    }
+
+    static OperationStepHandler checkNoOtherSibling(final String childType) {
+        return new OperationStepHandler() {
+            @Override
+            public void execute(OperationContext context, ModelNode operation) throws OperationFailedException {
+                PathAddress currentAddress = PathAddress.pathAddress(operation.get(ModelDescriptionConstants.OP_ADDR));
+                PathAddress parentAddress = currentAddress.subAddress(0, currentAddress.size() - 1);
+                Resource parent = context.readResourceFromRoot(parentAddress);
```

Comment:
Not a blocker, but this should still work with context.readResourceFromRoot(parentAddress, false); and it saves cloning.